src/audio/pipeline.c

```diff
@@ -336,7 +337,8 @@ int pipeline_buffer_connect(struct comp_buffer *source_buffer,
 			    struct comp_dev *sink_comp)
 {
 	trace_pipe("pipeline: connect source buffer %d -> sink comp %d",
-		   source_buffer->ipc_buffer.comp.id, sink_comp->comp.id);
+		   source_buffer ? source_buffer->ipc_buffer.comp.id : -1,
+		   sink_comp ? sink_comp->comp.id : -1);
```

Comment:
Aren't these checked by the caller in IPC handler ?

Reply:
I think that we only check struct ipc_comp pointers. We does not check cd/cb/pipeline pointers in this struct. 
```
struct ipc_comp_dev *icd_source;
struct ipc_comp_dev *icd_sink;

icd_source = ipc_get_comp(ipc, connect->source_id);
	if (icd_source == NULL) {
		trace_ipc_error("eCr");
		trace_error_value(connect->source_id);
		return -EINVAL;
	}

	icd_sink = ipc_get_comp(ipc, connect->sink_id);
	if (icd_sink == NULL) {
		trace_ipc_error("eCn");
		trace_error_value(connect->sink_id);
		return -EINVAL;
	}
```